packages/orm/lib/orm.js

```diff
@@ -0,0 +1,235 @@
+const chunkArray = (array, chunkSize) => {
+  if (chunkSize <= 0) chunkSize = 1;
+
+  return array.reduce((accum, item, index) => {
+    const chunkIndex = Math.floor(index / chunkSize);
+
+    if (!accum[chunkIndex]) {
+      accum[chunkIndex] = [];
+    }
+
+    accum[chunkIndex].push(item);
+
+    return accum;
+  }, []);
+};
+
+const runQuery = async ({
+  query,
+  variables,
+  keystone,
+  schemaName = 'public',
+  extraContext = {},
+}) => {
+  const { data, errors } = await keystone.executeGraphQL({
+    context: {
+      ...keystone.createContext({ schemaName, authentication: {}, skipAccessControl: true }),
+      ...extraContext,
+    },
+    query,
+    variables,
+  });
+  if (errors) throw errors[0];
+  return data;
+};
+
+const runPaginatedQuery = async ({
+  pageSize = 1,
+  variables = {},
+  verbose = false,
+  where = {},
+  ...rest
+}) => {
+  let skip = 0;
+  let latestResult;
+  const allItems = [];
+
+  if (verbose) {
+    console.log(`Started paginated query`);
+    console.log({ pageSize });
+  }
+
+  do {
```

Comment:
Is a do...while the best way to do this? It's probably fine but have you considered GOTO? :trollface: